Compile warnings with "invalid conversion from 'int' to 'collar_mode'"

C:\Program Files (x86)\Arduino\libraries\collar\CollarRx.cpp: In member function 'void CollarRx::buffer_to_collar_message(const uint8_t*, collar_message*)':
C:\Program Files (x86)\Arduino\libraries\collar\CollarRx.cpp:36:42: warning: invalid conversion from 'int' to 'collar_mode' [-fpermissive]
   msg->mode     = (collar_mode)buffer[2] & 0x0F;
                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
C:\Program Files (x86)\Arduino\libraries\collar\CollarRx.cpp:37:54: warning: invalid conversion from 'int' to 'collar_channel' [-fpermissive]
   msg->channel  = (collar_channel)(buffer[2] & 0xF0) >> 4;
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~

--- a/collar/CollarRx.cpp
+++ b/collar/CollarRx.cpp
@@ -33,8 +33,8 @@ void CollarRx::buffer_to_collar_message(const uint8_t buffer[5], struct collar_m
   memcpy(&msg->id, buffer, 2);
 
   // byte 2 = mode & channel
-  msg->mode     = (collar_mode)buffer[2] & 0x0F;
-  msg->channel  = (collar_channel)(buffer[2] & 0xF0) >> 4;
+  msg->mode     = (collar_mode)(buffer[2] & 0x0F);
+  msg->channel  = (collar_channel)((buffer[2] & 0xF0) >> 4);
 
   // byte 3 = power
   msg->power    =  buffer[3];

